tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/util/PingThread.java

```diff
@@ -84,15 +84,17 @@ protected void ping() {
 
 				URL pingUrl = new URL(url);
 				URLConnection conn = pingUrl.openConnection();
+
 				((HttpURLConnection)conn).setInstanceFollowRedirects(false);
-				((HttpURLConnection)conn).getResponseCode();
+				int code = ((HttpURLConnection)conn).getResponseCode();
 
 				// ping worked - server is up
-				if (!stop) {
+				if (!stop && code == 200) {
 					Thread.sleep(200);
 					behaviour.setServerStarted();
+					stop = true;
 				}
-				stop = true;
+				Thread.sleep(1000);
```

Comment:
Why are you sleeping for 1 second here if the server responded true?  Should this is be in an `else` block?